doc/api/esm.md

```diff
@@ -159,6 +156,15 @@ via the paths defined in [`"exports"`][].
 For details on these package resolution rules that apply to bare specifiers in
 the Node.js module resolution, see the [packages documentation](packages.md).
 
+### Mandatory file extensions
+
+A file extension must be provided when using the `import` keyword to resolve
+relative or absolute specifiers. Directory indexes (e.g. `'./startup/index.js'`)
+must also be fully specified.
```

Comment:
this is only true for relative paths; anything under a bare specifier isn't a path and thus has no extension requirement.

Reply:
Same as below, it's a part of current documentation that got reverted. 😅